airflow/providers/ftp/hooks/ftp.py

```diff
@@ -182,10 +185,12 @@ def write_to_file_with_progress(data):
             callback = output_handle.write
 
         remote_path, remote_file_name = os.path.split(remote_full_path)
+        current_path = conn.pwd()
         conn.cwd(remote_path)
         self.log.info("Retrieving file from FTP: %s", remote_full_path)
         conn.retrbinary(f"RETR {remote_file_name}", callback, block_size)
```

Comment:
Here you can provide the path of the file instead of its name to the command.

Reply:
You're right. The old implementation confused me, I thought that this was impossible - so I tried to get around it by calling the method again. Perhaps there was a reason for this approach?

https://github.com/apache/airflow/pull/35105/commits/86f92cabba9156bbe1dbce1aa46bc33fce901e27